Base: Add Keyboard Shortcuts & Tips and Tricks man pages + Keyboard symbols (#17672)
SerenityOS/serenity: Base/usr/share/man/man7/Tips-and-Tricks.md

![Welcome Icon](/res/icons/32x32/app-welcome.png)

## Name
Tips and Tricks

## Description
This is a list of useful tips and tricks to help you make the most out of SerenityOS.

## General

> * When on the Desktop or in File Manager, start typing and the nearest match will be selected.

Reviewer: itʼd be good to use one of the “active” or “focus” words here for more clarity. 😉

```suggestion
* […], start typing to select the nearest match.
```

to be frank, “the nearest match” is not the best wording… any ideas to be just slightly more specific here? 🤔

Author: I didn't want to use the world "search" because that has wider connotations and felt I'd found a good solution.
As you type only the icons/documents in your current focus (e.g. on the desktop or in the current folder in the File Manager) are being searched through.
I think your amendment makes it read much better though, so I will change it to that!

Author: Or "When on the Desktop or in File Manager, start typing to select the nearest match shortcut or filename"

* Bold text in context menus hints at the default behavior of a double-click.
* Hold `Ctrl` to accelerate mouse wheel interaction with sliders and spin boxes.
* Hold `Ctrl` while activating a menu item to prevent that menu from closing.

> * Many applications will open a compatible file if you drag-and-drop it into their window.

Reviewer: “compatible” is not the best here imho, it begs question for compatibility with what, and itʼs not really obvious that it is about the application itself; maybe this could be slightly more precise, e.g.  
```suggestion
* Many applications will open a file they can handle if you drag-and-drop it into their window.
```
i believe that the suggestion above is somewhat wrong and you could provide a better text for that hint. 🫣

Author: I've tried various alternatives ("will open a _relevant_ file?") but still think the original version works the best.
I think it's obvious you can drag a `.txt` file into Text Editor.

* Default file and protocol associations can be changed in `~/.config/LaunchServer.ini`.

### Window Management
* Double-click a window's title bar to maximize it.

> * Click on a window's icon to open the context menu.

Reviewer: i believe thatʼs not the context menu but the window menu (or at least should be—please double check that!).

* Double-click on the edge of an application's window to maximize it in that direction.
* Middle-click on a window's maximize button to extend the window vertically (this can be undone in the same way).
* Resizable windows can be dragged to all sides or corners of the screen, where they will automatically resize to fill half or one-quarter of the screen.
* Right-click on the Workspace Picker applet and select 'Workspace Settings' to easily customize the number and layout of Workspaces (virtual desktops).

### Fun
* It can help to get a second pair of `$ Eyes` on a problem… or fifty: `$ Eyes -n 100`.

## Applications

### [Assistant](help://man/1/Applications/Assistant)

> * Assistant can help you to quickly find files and applications. Summon it with `Super+Space`.

Reviewer: (“summon” is a word i really like but maybe this should be more of simple english—an “open” could be better here?)

Author: “Launch” or “Open” is probably the right word.
I liked “Summon” because it suggests Assistant is always available in the background ready to appear at a moments notice (though sounds a little occult).
I agree we should go with plain English rather than trying to be elegant.

Reviewer: “bring up” is another thing that comes to my mind; but if we open dialog windows, then why not the assistant? “launch” is a synonym for “run” or “execute” for me—this would contradict your point about assistant being ready in the background (on the other hand, thatʼs a technicality which does not really matter for the end user 🤔).

(i believe that technically weʼre opening assistantʼs _window_. you could try to go around that if you want—pretty simple and very accurate).

Author: "Open" it is 🙂

> * Assistant can help you to quickly find files and applications. Summon it with `Super+Space`.

Reviewer: not really sure with that tip… it feels detached from the second one. i believe all tips should be standalone.

the first tip says in fact what the assistant is and how to open it with a keystroke…

hard to tell here, really!

i think that the second tip is a real tip; maybe you could rephrase the tips, so you could tell:
* how to open the assistant (dubious?)
* that you can open applications (more dubious?)
* that you can find files and open them in the default application
* that you can open urls in the default application (e.g. Browser)
* that can do simple computations in it (i believe itʼs possible)

files and urls could possibly be merged as those are virtually the same in some way…? 🤔

Author: I'm trying to keep this short and sweet:

- The first tip quickly tells you there is a system search/launcher and how to access it.
- The second tips lists the only other functionality (that I know of).

I also thought you could do simple sums in Assistant, but have never managed it. Did we both dream about this functionality or was there a PR once upon a time (maybe it was never merged)?

* You can enter a URL for opening in the Browser.

### [Browser](help://man/1/Applications/Browser)

> * Browser has built-in ad blocking. Filter content by adding new domains to `~/.config/BrowserContentFiltering.txt`.

Reviewer: ```suggestion
* Browser has built-in content filtering, which can be used for ad blocking. Update the filters in `~/.config/BrowserContentFiltering.txt`.
```
or something like that? my case is to point that it is the content filtering which is built in, not the ad blocking! 😉 (which might be a broader thing.)

### Keyboard Mapper

> * Custom keymaps can be created and edited with `$ KeyboardMapper`.

Reviewer: active voice?
```suggestion
* Create and edit custom keymaps with `$ KeyboardMapper`.
```

### Run
* The Run dialog accepts all [Shell](help://man/5/Shell) commands.

### [Terminal](help://man/1/Applications/Terminal)
* Bold or underlined text in Terminal can be double-clicked to open or right-clicked for additional actions.
* Many Serenity applications already have convenient aliases. Use `$ cat /etc/shellrc` to view them.

### [Text Editor](help://man/1/Applications/TextEditor)
* Text files can be dragged directly from Terminal and dropped on Text Editor to open them.
* Text Editor has multiple viewing modes. You can edit HTML or Markdown and live preview it at the same time.

## Development
* Supplying `# profile` with a PID of `-1` as root enables systemwide profiling.

> * Applications can be viewed with [Inspector](help://man/1/Applications/Inspector), which facilitates process inspection via Remote Procedure Call (RPC), by including `MAKE_INSPECTABLE=1` in their environment. For example, to make the Eyes application inspectable you would enter the following Shell commands:
> ```sh
> $ export MAKE_INSPECTABLE=1
> $ Eyes &
> ```

Reviewer: not sure if itʼs entirely correct, but iʼd suggest the following
```suggestion
* Make applications inspectable by including `MAKE_INSPECTABLE=1` in their environment. Inspectable processes can be examined with [Inspector](help://man/1/Applications/Inspector) via Remote Procedure Calls (RPCs). For example, to make the Eyes application inspectable, enter the following Shell command:
```
```sh
$ MAKE_INSPECTABLE=1 Eyes &
```

Author: This document is based on [tips.txt](https://github.com/SerenityOS/serenity/blob/f27f2e0d3b3fe08e68534676a58aab8e1eeea676/Base/home/anon/Documents/tips.txt) hence why this part is included even though I don't fully understand it.
I asked for help on #development and [Andrew Kaster taught me](https://discord.com/channels/830522505605283862/830739873119207426/1075928327711359036) how to actually use the command, so that's what I included. I tried your command and it didn't work for me - but I bow to you on this one, for I know not.

Author: After testing multiple times, I've reverted to the original commands as they're the only ones that work:
```sh
$ export MAKE_INSPECTABLE=1
$ Eyes &
```

* Hold `Ctrl` while activating a menu item to prevent that menu from closing.
* Many applications can open a compatible file if you drag-and-drop it into their window.
* Change default file and protocol associations in `~/.config/LaunchServer.ini`.

### Window Management
* Double-click a window's title bar to maximize it.
* Click on a window's icon to open the window's context menu.
* Double-click on the edge of an application's window to maximize it in that direction.
* Middle-click on a window's maximize button to extend the window vertically (this can be undone in the same way).
* Drag resizable windows to any side or corner of the screen to automatically resize them to fill half or one-quarter of the screen.
* Right-click on the Workspace Picker applet and select 'Workspace Settings' to easily customize the number and layout of Workspaces (virtual desktops).

### Fun
* It can help to get a second pair of `$ Eyes` on a problem… or fifty: `$ Eyes -n 100`.

## Applications

### [Assistant](help://man/1/Applications/Assistant)
* Assistant can help you to quickly find files and launch applications. Open it with `Super+Space`.
* Enter a URL to open it in the web browser.

> * Perform quick calculations by prefixing your expression with the equal sign (`=`), e.g. `=2*5`.

Reviewer: something along the lines `Evaluate simple mathematical expressions by prepending them with the equality sign`?

prefixes are strictly part of what your attaching them to; prepending separates this equality-sign computation marker from the expression logically—just as you put that in words. 🤔

need not to mimic that 100%, take it with a grain of salt (a bigger one) as i am not a native speaker! 🥸

>     * Pressing Return will copy the result to the clipboard.

Reviewer: i feel you can merge it with a previous (use full stop or semicolon to separate?).

### [Browser](help://man/1/Applications/Browser)
* Browser has built-in content filtering, which can be used for ad blocking. Update the filters in `~/.config/BrowserContentFiltering.txt`.

### Keyboard Mapper
* Create and edit custom keymaps with `$ KeyboardMapper`.

### Run
* The Run dialog accepts all [Shell](help://man/5/Shell) commands.

### [Terminal](help://man/1/Applications/Terminal)
* Some of the bold or underlined text in Terminal can be double-clicked to open or right-clicked for additional actions.

>     * For example, right-click on a file or folder and select 'Copy URL' to copy the file path.

Reviewer: that would be a `file or folder path` to be precise. 😉 (“to copy it.”; not sure if mentioning the clipboard is necessary here)
